sensor/common/registry/registry_store.go

```diff
@@ -392,3 +393,220 @@ func (rs *Store) GetMatchingCentralRegistryIntegrations(imgName *storage.ImageNa
 
 	return regs
 }
+
+// UpsertSecret upserts a pull secret into the store.
+func (rs *Store) UpsertSecret(namespace, secretName string, dockerConfig config.DockerConfig, serviceAcctName string) {
+	if !features.SensorPullSecretsByName.Enabled() {
+		rs.upsertSecretByHost(namespace, secretName, dockerConfig, serviceAcctName)
+		return
+	}
+
+	rs.upsertSecretByName(namespace, secretName, dockerConfig, serviceAcctName)
+}
+
+func (rs *Store) upsertSecretByHost(namespace, secretName string, dockerConfig config.DockerConfig, serviceAcctName string) {
+	isGlobalPullSecret := namespace == openshiftConfigNamespace && secretName == openshiftConfigPullSecret
+	fromDefaultSA := serviceAcctName == defaultSA
+
+	for registryAddress, dce := range dockerConfig {
+		registryAddr := strings.TrimSpace(registryAddress)
+
+		if fromDefaultSA {
+			// We assume that registries found in the dockercfg secret managed by OCP for the default
+			// service account only references hostnames for the OCP internal registry.
```

Comment:
Is this true? I know originally that's all we cared about, but I don't recall if this is a true statement

Reply:
Perhaps you tell me 😉 , that comment was mostly taken from the old `secrets.go`. 

As far as I know/have seen - yes, each SA has a unique token for accessing the OCP internal registry. When a service account is created, a `*-dockercfg-*` secret is also created, on OCP the hosts in that secret align with the internal registry, the docs [seem to indicate](https://docs.openshift.com/container-platform/4.16/authentication/using-service-accounts-in-applications.html#auto-generated-sa-token-secrets_using-service-accounts) that is the purpose of the `dockercfg` secret as well.

```
$ k get secret default-dockercfg-hqlwr -o json | jq -r '.data.".dockercfg"' | base64 -d | jq 'keys'
[
  "172.30.239.146:5000",
  "image-registry.openshift-image-registry.svc.cluster.local:5000",
  "image-registry.openshift-image-registry.svc:5000"
]
``` 
I've tried adding extra hosts to this secret and although the updated secret is accepted by the api server, it is immediately corrected / the extra data removed.